File: src/main/java/com/sfh/pokeRogueBot/model/modifier/impl/BerryModifierItem.java
package com.sfh.pokeRogueBot.model.modifier.impl;

import com.sfh.pokeRogueBot.model.enums.BerryType;
import com.sfh.pokeRogueBot.model.modifier.GeneratedPersistentModifierType;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class BerryModifierItem extends PokemonHeldItemModifierItem implements GeneratedPersistentModifierType {

    public static final String TARGET = "BerryModifierType";

    private BerryType berry;

}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/modifier/impl/BerryModifierItem.kt
package com.sfh.pokeRogueBot.model.modifier.impl

import com.sfh.pokeRogueBot.model.enums.BerryType
import com.sfh.pokeRogueBot.model.modifier.GeneratedPersistentModifierType

class BerryModifierItem : PokemonHeldItemModifierItem(), GeneratedPersistentModifierType {

    companion object {
        const val TARGET = "BerryModifierType"
    }

    var berry: BerryType? = null
}